src/shared/behaviors/collection.html

```diff
@@ -0,0 +1,206 @@
+<link rel="import" href="../js/datautils.html"/>
+<link rel="import" href="../js/behaviorutils.html"/>
+<script>
+/**
+ * @license
+ * Copyright (c) 2015 MediaMath Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at http://mediamath.github.io/strand/LICENSE.txt
+
+*/
+(function (scope) {
+
+	var DataUtils = StrandLib.DataUtils;
+	var BehaviorUtils = StrandLib.BehaviorUtils;
+
+	function _generateCid(collection, model) {
+		model.cId = collection._cidIndex++;
+	}
+
+	var _eventPrefix = "data-";
+
+	scope.Collection = {
+
+		properties:{
+			_cidIndex:{
+				type:Number,
+				value:0
+			},
+			added:{
+				type:Array,
+				value: function() {
+					return [];
+				},
+				notify:true
+			},
+			removed:{
+				type:Array,
+				value: function() {
+					return [];
+				},
+				notify:true
+			},
+			modified:{
+				type:Array,
+				value: function() {
+					return [];
+				},
+				notify:true
+			}
+		},
+
+		observers:[
+			'_dataArrayChanged(data.splices)',
+			'_deepChanged(data.*)'
+		],
+
+		add: function(model, silent, force) {
+			this._silent = silent;
+
+			if (!model.cId || model.cId === -1) {
+				_generateCid(this, model);
+			}
+			var check = this.data.filter(function(m) {
+				if (m.cId && model.cId) { return m.cId === model.cId; }
+				if (m.mId && model.mId) { return m.mId === model.mId; }
```

Comment:
Is the above intentionally preferred over something like:

```
if (m.cId && model.cId && m.cId === model.cId) { return true; }
if (m.mId && model.mId && m.mId === model.mId) { return true; }
```

In other words(, and although it may be a somewhat exceptional edge-case), the question is "provided `.cId` exists, is a `.cId` mismatch strictly more important than a `.mId`match or not?"